src/components/reusable/SearchBar.js

```diff
@@ -0,0 +1,132 @@
+    if (event.key === "ArrowUp" && highlightedIndex > 0) {
+      // Ensure page is not re-rendered
+      event.preventDefault();
+      setHighlightedIndex(highlightedIndex - 1);
+    }
+    // Action for when a user hits the down arrow, will increase the highlighted index
+    if (
+      event.key === "ArrowDown" &&
+      highlightedIndex < searchResults.length - 1
+    ) {
+      // Ensure page is not re-rendered
+      event.preventDefault();
+      setHighlightedIndex(highlightedIndex + 1);
+    }
+  };
+
+  // Keyboard input in search box
+  const handleTextChange = (event) => {
+    // If a user clears everything in the text box then the drop down should be closed
+    if(event.target.value === ""){
+      closeDropdown();
+    }
+    event.preventDefault();
+    // Update the search value to the new text
+    setSearchValue(event.target.value);
+  };
+
+  // Handles clicking on dropdown options
+  const handleClick = (value) => {
+      // Call the search callback and close the dropdown
+      selectItem(value);
+      closeDropdown()
+    };
+
+  /* Called anytime the drop down needs to be closed
+  Will hid the results, restore highlighted index, and set the search string to empty*/
+  const closeDropdown = () => {
+    setShowResults(false);
+    setHighlightedIndex(-1);
+    setSearchResults([]);
+    setSearchValue("")
+  }
+
+  // Close search upon clicking anywhere outside of the component
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (searchRef.current && !searchRef.current.contains(event.target)) {
+        // Close the dropdown
+        closeDropdown();
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [searchRef]);
+
+  return (
+    <div ref={searchRef} className="relative">
+      <div className="flex border border-another-gray rounded-sm">
+        <button className="px-4 bg-dark-gray rounded-md">
+          <FaSearch className="text-white" />
+        </button>
+        <input
+          className="bg-dark-gray px-4 py-2 w-64 rounded-md text-white focus:outline-none"
+          type="text"
+          placeholder="Search for stocks"
+          onChange={handleTextChange}
+          onKeyDown={handleKey}
+          value={searchValue}
+        />
+      </div>
+      {showResults && searchValue.length > 0 && (
+        <div className="absolute bg-darker-gray rounded-sm shadow-lg text-white w-full">
+          {searchResults.map((result, index) => (
+            <p
+              className={`px-4 py-2 hover:bg-another-gray cursor-pointer ${
+                highlightedIndex === index
+                  ? "bg-another-gray border-l border-light-gray"
+                  : ""
+              }`}
+              key={result}
+              onClick={() => handleClick(result)}
+              onKeyDown={() => handleKey(result)}
```

Comment:
is this needed? currently this function is already being done in the input box, those <p/Ps are not the ones that are getting new values for search.